src/test/java/com/epam/mjc/collections/map/KeyValueSwapperTest.java

```diff
@@ -0,0 +1,64 @@
+package com.epam.mjc.collections.map;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class KeyValueSwapperTest {
+    public static Stream<Arguments> testCases() {
+        return Stream.of(
+                Arguments.of("EmptyMap",
+                        Map.of(),
+                        Map.of()),
+
+                Arguments.of("NonEmptyMap",
+                        Map.of(1, "Monday",
+                                2, "Tuesday",
+                                3, "Wednesday",
+                                4, "Thursday",
+                                5, "Friday",
+                                6, "Saturday",
+                                7, "Sunday"),
+                        Map.of("Monday", 1,
+                                "Tuesday", 2,
+                                "Wednesday", 3,
+                                "Thursday", 4,
+                                "Friday", 5,
+                                "Saturday", 6,
+                                "Sunday", 7)),
+
+                Arguments.of("MapWithSameValues",
+                        Map.of(1, "Monday",
+                                2, "Tuesday",
+                                3, "Wednesday",
+                                4, "Thursday",
+                                5, "Friday",
+                                6, "Saturday",
+                                7, "Sunday",
+                                10, "Friday"),
+                        Map.of("Monday", 1,
+                                "Tuesday", 2,
+                                "Wednesday", 3,
+                                "Thursday", 4,
+                                "Friday", 5,
```

Comment:
Friday 5 or Friday 10?

Reply:
_If in the source map `Map<Integer, String>` there are the same values for different keys, in the
new `Map<String, Integer>` this value should map to a **smaller** key value._
So, 5<10 => Friday mapped to 5

Reply:
Okay, thanks for the clarifying